tests/containers/bci_prepare.pm

```diff
@@ -134,6 +134,16 @@ sub run {
             # Note: Both lines are aligned, thus the additional space
             record_info('builds', "CONTAINER_IMAGE_BUILD:  $build\norg.opensuse.reference: $reference");
             die('Missmatch in image build number. The image build number is different than the one triggered by the container bot!') if ($reference !~ /$build$/);
+            if (check_var('BCI_TEST_METADATA', '1')) {
+                # Check if we can run Metadata tests when the variable BCI_TEST_METADATA is set.
+                # Metadata tests compare the image under tests vs the released image in the public registry,
+                # so we need to make sure the public image exists, which won't be the case for brand new images
```

Comment:
This is wrong. The metadata tests check the labels of the image and skipping them universally is a **bad** idea.

Reply:
Not universally, they will be skipped for images that are not public yet. Once they are published in public repo, the check will be `true` and it will run the metadata tests.

Reply:
Yes and then we will figure out that the metadata are wrong, but the damage has been already done. What you actually want to do, is just skip the test of the reference not all of them

Reply:
ok, I guess we can handle that condition in the bci-tests repo instead. I'll revert

Reply:
we don't have to handle it in the bci-tests repo, you can also just filter the test out via by appending `-k "not reference"` to the tox call

Reply:
Sure, but then we might also have failing tests in the bci-tests CI when some image is not yet released but enabled. If that's ok, then I'll add that option in this code.

Reply:
If you have a good & reliable way how to do that, please send a PR!